### DISCUSSION TITLE
what different nanoid with Crypto.randomUUID()

### DISCUSSION THREAD

**Discussion by vaynevayne** [ai/nanoid#492] [Q&A]
Now that you have native methods, do you no longer need a nanoid?

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

**Comment by ai** [comment]
You may need Nano ID to have shorter IDs

**Comment by teetotum** [comment]
@ai As far as I understand there is a restriction with the new web crypto API that it can only be used when a page is loaded via HTTPS and not HTTP.
Am I correct to assume that the same restriction therefore exists for Nano-ID?
What would happen if I called `nanoid()` and the current page runs under http?

> Secure context: This feature is available only in [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (HTTPS), in some or all [supporting browsers](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID#browser_compatibility).
From https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID
